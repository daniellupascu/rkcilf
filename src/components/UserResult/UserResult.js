import React from "react";
import PropTypes from "prop-types";

import Loader from "../Loader/Loader";
import UserInformation from "../UserInformation/UserInformation";
import { LoaderContainer } from "./UserResultStyled";

const UserResult = ({ isLoading, isNotFound, user }) => {
  return (
    <>
      {isLoading ? (
        <LoaderContainer>
          <Loader />
        </LoaderContainer>
      ) : isNotFound ? (
        <LoaderContainer>user not found</LoaderContainer>
      ) : user.id ? (
        <UserInformation user={user} />
      ) : null}
    </>
  );
};

export default UserResult;

UserResult.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isNotFound: PropTypes.bool.isRequired,
  user: PropTypes.object,
};
