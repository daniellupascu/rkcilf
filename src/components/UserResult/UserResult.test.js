import React from "react";
import renderer from "react-test-renderer";
import UserResult from "./UserResult";

describe("Test UserResult Component", () => {
  it("renders correctly", () => {
    const user = {
      name: "Frank Johnson",
      html_url: "https://github.com/",
      blog: "https://github.com/",
      bio: "users bio",
      location: "Arizona",
      public_repos: 10,
      public_gists: 0,
      avatar_url:
        "https://github.githubassets.com/images/modules/logos_page/Octocat.png",
    };
    const tree = renderer
      .create(<UserResult user={user} isLoading={false} isNotFound={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly the loading state", () => {
    const tree = renderer
      .create(<UserResult user={{}} isLoading={true} isNotFound={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly the not found state", () => {
    const tree = renderer
      .create(<UserResult user={{}} isLoading={false} isNotFound={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
