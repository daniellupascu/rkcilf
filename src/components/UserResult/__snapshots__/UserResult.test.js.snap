// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserResult Component renders correctly 1`] = `null`;

exports[`Test UserResult Component renders correctly the loading state 1`] = `null`;

exports[`Test UserResult Component renders correctly the not found state 1`] = `null`;
